source/parsing/Token.cpp

```diff
@@ -153,6 +153,22 @@ span<Token const> Trivia::getSkippedTokens() const {
     return {tokens.ptr, tokens.len};
 }
 
+Trivia Trivia::deepClone(BumpAllocator& alloc) const {
```

Comment:
I am confused about this function. The point is to clone the Trivia so that it's exactly the same, just in a new object yes? You need to check what kind of trivia it is and clone it appropriately, not always force it to be a full location trivia. The result should only be a full location trivia if the source trivia also has a full location (which is pretty rare btw). 

Basically you want to clone whichever member of the union is valid, which you can know by looking at the `kind` member.

Reply:
So something like this: 
```
Trivia Trivia::clone(BumpAllocator& alloc) const {
    Trivia result;
    result.kind = kind;

    switch (kind) {
        case TriviaKind::Directive:
        case TriviaKind::SkippedSyntax:
            result.syntaxNode = syntaxNode;
            break;
        case TriviaKind::SkippedTokens:
            result.tokens = tokens;
            break;
        default:
            if (hasFullLocation) {
                result.fullLocation = alloc.emplace<FullLocation>();
                result.fullLocation->text = fullLocation->text;
                result.fullLocation->location = fullLocation->location;
                result.hasFullLocation = true;
            }
            else {
                result.rawText = rawText;
            }
            break;
    }

    return result;
}
```